app/Http/Services/PurchaseOrderService.php

```diff
@@ -74,4 +74,25 @@ private static function generatePoNumber()
         $newNumber = str_pad($lastRefNo + 1, 6, '0', STR_PAD_LEFT);
         return "PO-{$year}-{$newNumber}";
     }
+
+    public function applyFilters($query)
+    {
+        if (request()->has('rs_number')) {
+            $query->where('rs_number', 'like', '%' . request()->input('rs_number') . '%');
+        }
+        if (request()->has('po_number')) {
+            $query->where('po_number', request()->input('po_number'));
+        }
+        if (request()->has('transaction_date')) {
+            $query->where('transaction_date', 'like', '%' . request()->input('transaction_date') . '%');
+        }
+        return $query;
+    }
+
+    public function getAllRequest()
+    {
+        $query = RequestPurchaseOrder::orderBy('created_at', 'DESC');
+        $query = $this->applyFilters($query);
+        return $query->paginate(config('app.pagination.per_page', 15));
+    }
```

Comment:
murag wala man ni nagamit mark? ayaw na pud i service ang pag get sa request
services is for transactions lang